extensions/src/org/pentaho/platform/plugin/services/security/userrole/ldap/DefaultLdapUserRoleListService.java

```diff
@@ -114,6 +117,13 @@ public void afterPropertiesSet() throws Exception {
         roles.add( roleString );
       }
     }
+    // Now add extra role if it does not exist in the list
+    for ( String extraRole : extraRoles ) {
+      if ( !roles.contains( extraRole ) ) {
```

Comment:
extra roles are "Authenticated" and "Anonymous" in our platform. The user has an option of adding to that list and will be added to the role retrieved from the third party security provider. In this case we will mostly have "Authenticated" as an extra role.